kafka/partitions.go

```diff
@@ -21,25 +23,47 @@ Iter:
 	return diff
 }
 
-func GetPartitions(client sarama.Client, topics []string) ([]int32, error) {
-	partitionCount := 0
-	partitions := make([]int32, 0)
-	var err error
+func GetPartitions(client *confluent.Consumer, topics []string, retries, backoff, timeout int) ([]int32, error) {
+	var partitions []int32
+	var ok bool
+	var tm confluent.TopicMetadata
 	for i, topic := range topics {
-		partitions, err = client.Partitions(topic)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to get partitions for topic %s. %s", topic, err)
-		}
-		if len(partitions) == 0 {
-			return nil, fmt.Errorf("No partitions returned for topic %s", topic)
-		}
-		if i > 0 {
-			if len(partitions) != partitionCount {
-				return nil, fmt.Errorf("Configured topics have different partition counts, this is not supported")
+		for retry := retries; retry > 0; retry-- {
+			metadata, err := client.GetMetadata(&topic, false, timeout)
+			if err != nil {
+				log.Warn("kafka: failed to get metadata from kafka client. %s, %d retries", err, retry)
+				time.Sleep(time.Duration(backoff) * time.Millisecond)
+				continue
+			}
+
+			// if kafka's auto.create.topics is enabled (default) then a topic will get created with the default
+			// settings after our first GetMetadata call for it. But because the topic creation can take a moment
+			// we'll need to retry a fraction of second later in order to actually get the according metadata.
+			if tm, ok := metadata.Topics[topic]; !ok || tm.Error.Code() == confluent.ErrUnknownTopic {
+				log.Warn("kafka: unknown topic %s, %d retries", topic, retry)
+				time.Sleep(time.Duration(backoff) * time.Millisecond)
+				continue
+			}
+
+			if tm, ok = metadata.Topics[topic]; !ok || len(tm.Partitions) == 0 {
```

Comment:
no need to check if topic is in metadata.Topics, it has already been checked.

Maybe change these 2 checks into
```
tm, ok := metadata.Topics[topic]
if !ok || tm.Error.Code() == confluent.ErrUnknownTopic {
  log.Warn("kafka: unknown topic %s, %d retries", topic, retry)
  time.Sleep()
  continue
}
if len(tm.Partitions) == 0 {
  log.Warn("kafka: 0 partitions returned for %s, %d retries left, %d backoffMs", topic, retry, backoff)
  sleep()
  continue
}
```